File: backend/src/InsuraTech.Domain/Claims/ClaimStatusHistory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using InsuraTech.Domain.Common;

namespace InsuraTech.Domain.Claims
{
    public sealed class ClaimStatusHistory : Entity
    {
        public Guid ClaimId { get; private set; }
        public ClaimStatus Status { get; private set; }
        public string ResponsibleUser { get; private set; } = null!;
        public string? Observations { get; private set; }
        public DateTime ChangedAt { get; private set; }
        private ClaimStatusHistory() { }

        public static ClaimStatusHistory Create(
            Guid claimId, ClaimStatus status, string responsibleUser, string? observation = null)
        { 
            if(string.IsNullOrWhiteSpace(responsibleUser))
                throw new ArgumentNullException("Responsable user is requiered. ", nameof(responsibleUser));
            return new ClaimStatusHistory
            {
                ClaimId = claimId,
                Status = status,
                ResponsibleUser = responsibleUser,
                ChangedAt = DateTime.UtcNow,
            };
        }


    }
}

File: backend_v2/src/main/java/com/insuratech/domain/claims/ClaimStatusHistory.java
package com.insuratech.domain.claims;

import lombok.Getter;

import java.time.Instant;

@Getter
public class ClaimStatusHistory {

    private final ClaimStatus status;
    private final Instant changedAt;
    private final String changedBy;
    private final String notes;

    public ClaimStatusHistory(ClaimStatus status, String changedBy, String notes) {
        this.status = status;
        this.changedAt = Instant.now();
        this.changedBy = changedBy;
        this.notes = notes;
    }
}
